circuit/src/tables.rs

```diff
@@ -222,6 +258,35 @@ impl<
         let mul_trace = self.generate_mul_trace()?;
         let sub_trace = self.generate_sub_trace()?;
         let fake_merkle_trace = self.generate_fake_merkle_trace()?;
+        Ok(Traces {
+            witness_trace,
+            const_trace,
+            public_trace,
+            add_trace,
+            mul_trace,
+            sub_trace,
+            fake_merkle_trace,
+            sponge_trace: SpongeTrace::default(),
+        })
+    }
+
+    /// Run the circuit and generate traces
+    pub fn run_with_hash<P: CryptographicPermutation<[F; N]>, const N: usize>(
```

Comment:
Ugly, but I don't want to add the generics and bounds to the normal one.

More generally, it feels like it would be better to have a single `run` function handling all possible cases without all the generics. The only other solution is to somehow put all future generics and trait bounds in the CircuitBuilder/Circuit/CircuitRunner structs themselves, but it doesn't feel ideal either. Not sure how to do it properly.